jetty-util-ajax/src/main/java/org/eclipse/jetty/util/ajax/AsyncJSON.java

```diff
@@ -211,6 +213,19 @@ private boolean isCacheable()
     }
 
     private static final Object UNSET = new Object();
+    private static final Trie<Number> NUMBERS = new ArrayTrie<>();
+
+    static
+    {
+        for (String terminator : new String[] {" ", "\t", "\r", "\n", ",", "]", "}"})
+        {
+            NUMBERS.put("0" + terminator, 0L);
+            NUMBERS.put("1" + terminator, 1L);
+            NUMBERS.put("-1" + terminator, -1L);
+            NUMBERS.put("0.0" + terminator, 0.0D);
+            NUMBERS.put("1.0" + terminator, 1.0D);
+        }
```

Comment:
What on earth??

Reply:
I don't think there is evidence that this will make any difference.
I don't think CometD would benefit from this, for example, as the ack extension will immediately go to numbers > 1.
I don't have any other examples of typical JSON documents, but I would rather add a generic `put(String, Object)` and have applications to cache the common occurrences of their JSON documents.

Reply:
Doesn't matter what we think. It matters what the benchmarks say!
But yeah, these are probably low mileage optimisations, but then they are pretty simple and I wouldn't think they cost too much.

Note that numbers can't be done with a generic `Object` cache, because they don't have a natural terminator like `]`, `}` o `"`, so they need special handling to look for a number followed by something that is a non-number (@joakime does that explain your WOE!?).  although I think that probably only the `,` and perhaps ` ` terminators  would do.... the space saved would be better used adding 2...99 or something. Actually the space used to go to 999 might even be worth it.

Reply:
Hmmm a really non-objective survey of some json sources indicates that the empty array, empty map and empty string optimisations would rarely be used.
However there are lots of small numbers in the range of 0-99, so there might be some value to the NUMBERS approach.